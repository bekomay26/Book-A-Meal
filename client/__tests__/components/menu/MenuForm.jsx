import React from 'react';
import { shallow } from 'enzyme';
import MenuForm from '../../../src/components/menu/MenuForm';
import { getMealsResponse } from '../../__mocks__/mockMeal';

describe('Customer Order Form Component Test Suite', () => {
  let props;
  let wrapper;
  const mealArray = getMealsResponse.meals;

  beforeEach(() => {
    props = {
      goesWith: jest.fn().mockReturnValue([]),
      extras: jest.fn().mockReturnValue([]),
      onQtyChange: jest.fn(),
      extraStatus: '',
      handleChecked: jest.fn(),
      totalMealPrice: 0,
      onSave: jest.fn(),
      onClose: jest.fn(),
      saving: false,
      meal: mealArray[0],
    };
    wrapper = shallow(<MenuForm {...props} />);
  });

  it('renders without crashing', () => {
    expect(wrapper).toBeDefined();
    const subButton = wrapper.find('.orderButton');
    expect(subButton.prop('content')).toEqual('order');
  });

  it('renders without crashing when saving is true', () => {
    wrapper.setProps({ saving: true });
    expect(wrapper).toBeDefined();
    const subButton = wrapper.find('.orderButton');
    expect(subButton.prop('content')).toEqual('ordering');
  });
});
