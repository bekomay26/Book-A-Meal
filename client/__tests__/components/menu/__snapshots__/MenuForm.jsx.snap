// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Menu form Component should render menu form component correctly 1`] = `
<MenuForm
  extraStatus=""
  extras={
    [MockFunction] {
      "calls": Array [
        Array [
          Array [
            Object {
              "category": "OnTop",
              "id": 4,
              "price": 100,
              "title": "mutton",
            },
            Object {
              "category": "GoesWith",
              "id": 1,
              "price": 50,
              "title": "beef",
            },
            Object {
              "category": "OnTop",
              "id": 2,
              "price": 150,
              "title": "chicken",
            },
          ],
        ],
      ],
      "results": Array [
        Object {
          "isThrow": false,
          "value": Array [],
        },
      ],
    }
  }
  goesWith={
    [MockFunction] {
      "calls": Array [
        Array [
          Array [
            Object {
              "category": "OnTop",
              "id": 4,
              "price": 100,
              "title": "mutton",
            },
            Object {
              "category": "GoesWith",
              "id": 1,
              "price": 50,
              "title": "beef",
            },
            Object {
              "category": "OnTop",
              "id": 2,
              "price": 150,
              "title": "chicken",
            },
          ],
        ],
      ],
      "results": Array [
        Object {
          "isThrow": false,
          "value": Array [],
        },
      ],
    }
  }
  handleChecked={[MockFunction]}
  meal={
    Object {
      "createdAt": "2018-09-05T15:19:29.689Z",
      "description": null,
      "extras": Array [
        Object {
          "category": "OnTop",
          "id": 4,
          "price": 100,
          "title": "mutton",
        },
        Object {
          "category": "GoesWith",
          "id": 1,
          "price": 50,
          "title": "beef",
        },
        Object {
          "category": "OnTop",
          "id": 2,
          "price": 150,
          "title": "chicken",
        },
      ],
      "id": 54,
      "image_url": "https://res.cloudinary.com/dffiyhgto/image/upload/v1536160768/gfhgh980.jpg",
      "price": 980,
      "quantity": 1,
      "title": "gfhgh",
      "updatedAt": "2018-09-05T15:19:29.689Z",
    }
  }
  onClose={[MockFunction]}
  onQtyChange={[MockFunction]}
  onSave={[MockFunction]}
  saving={false}
  totalMealPrice={0}
>
  <div
    id="modal-container"
  >
    <Form
      as="form"
      className="menu-page-form"
      loading={false}
      onSubmit={[MockFunction]}
    >
      <form
        className="ui form menu-page-form"
        onSubmit={[Function]}
      >
        <div
          className="menu-page-form-body"
        >
          <img
            alt="meal"
            className="menu-page-meal-image"
            src="https://res.cloudinary.com/dffiyhgto/image/upload/v1536160768/gfhgh980.jpg"
          />
          <Header>
            <div
              className="ui header"
            >
              <div
                className="menu-page-meal-title col-10"
              >
                gfhgh
              </div>
              <div
                className="menu-page-meal-price col-2"
              >
                ₦
                980
              </div>
            </div>
          </Header>
          <div
            className="menu-page-meal-description"
          />
          <div
            className="menu-page-mealwith"
            id="mealwith"
          >
            <div
              className="row"
            >
              <h4
                className="col-6"
              >
                Goes With
              </h4>
              <h4
                className="col-3"
              >
                Qty
              </h4>
              <h4
                className="col-3 price-head"
              >
                Price
              </h4>
            </div>
          </div>
          <div
            className="menu-page-mealextra row hide-menu-extra"
            id="mealextras"
          >
            <div
              className="row"
            >
              <h4
                className="col-6"
              >
                Extra
              </h4>
              <h4
                className="col-3"
              >
                Qty
              </h4>
              <h4
                className="col-3 price-head"
              >
                Price
              </h4>
            </div>
          </div>
          <div
            className="pricetotal row"
          >
            <h4
              className="col-9"
            >
              Total Price
            </h4>
            <h4
              className="col-3 foodprice"
            >
              ₦
              0
            </h4>
          </div>
        </div>
        <FormButton
          as={[Function]}
          className="orderButton "
          content="order"
          control={[Function]}
          disabled={false}
          loading={false}
        >
          <FormField
            className="orderButton "
            content="order"
            control={[Function]}
            disabled={false}
            loading={false}
          >
            <div
              className="field orderButton "
            >
              <Button
                as="button"
                content="order"
                disabled={false}
                loading={false}
                role="button"
              >
                <button
                  className="ui button"
                  onClick={[Function]}
                  role="button"
                >
                  order
                </button>
              </Button>
            </div>
          </FormField>
        </FormButton>
      </form>
    </Form>
  </div>
</MenuForm>
`;
